### PR TITLE
Build images from scratch

### PR DESCRIPTION
**Pull request by polskikiel**
**Description**

Changes proposed in this pull request:

- Build images from scratch

**Related issue(s)**

### CODE REVIEW

File: resources/service-catalog/templates/tests/test.yaml [hunk 1]
```diff
@@ -39,6 +39,6 @@ spec:
         args:
         - "-c"
         #TODO: there's no curl in the image. It depends on "shareProcessNamespace=true" and may not work in all environments.
-        - "echo 'TESTING start'; sleep 10; ./entrypoint.sh; exit_code=$?; echo code is $exit_code; echo 'killing pilot-agent...'; pkill -INT pilot-agent; sleep 4; exit $exit_code;"
+        - "echo 'TESTING start'; sleep 10; ./entrypoint.sh; exit_code=$?; echo code is $exit_code; echo 'killing pilot-agent...'; curl -XPOST http://127.0.0.1:15020/quitquitquit; sleep 4; exit $exit_code;"
```
Comment (polskikiel) [inline resources/service-catalog/templates/tests/test.yaml]: I discussed it with goats. It's a new way to kill sidecar after the test

File: components/application-broker/Dockerfile [hunk 2]
```diff
@@ -6,16 +6,17 @@ ADD . $SRC_DIR
 WORKDIR $SRC_DIR
 
 RUN CGO_ENABLED=0 go build -o /application-broker ./cmd/broker
+RUN apk add --no-cache curl
 
-FROM alpine:3.10
-
+FROM alpine:3.10 as certs
```
Comment (mszostok) [inline components/application-broker/Dockerfile]: use latest ver

File: components/application-broker/Dockerfile [hunk 3]
```diff
@@ -6,16 +6,17 @@ ADD . $SRC_DIR
 WORKDIR $SRC_DIR
 
 RUN CGO_ENABLED=0 go build -o /application-broker ./cmd/broker
+RUN apk add --no-cache curl
```
Comment (mszostok) [inline components/application-broker/Dockerfile]: why u need that? it's not copied to the final image

File: components/service-binding-usage-controller/Dockerfile [hunk 4]
```diff
@@ -1,21 +1,30 @@
 FROM golang:1.12.10-alpine3.10 as builder
 
-RUN apk update && apk add curl
-
 ENV SRC_DIR=/go/src/github.com/kyma-project/kyma/components/service-binding-usage-controller
 ADD . $SRC_DIR
 
+RUN mkdir /user && \
```
Comment (mszostok) [inline components/service-binding-usage-controller/Dockerfile]: add business comment why we need that

File: components/service-binding-usage-controller/Dockerfile [hunk 5]
```diff
@@ -1,21 +1,30 @@
...
+    echo 'appuser:x:2000:' > /user/group
+RUN mkdir -p tmp
+
 WORKDIR $SRC_DIR
 
 RUN CGO_ENABLED=0 go build -o /binding-usage-controller ./cmd/controller
 
-FROM alpine:3.10
-
+FROM alpine:3.10 as certs
```
Comment (mszostok) [inline components/service-binding-usage-controller/Dockerfile]: use latest

File: resources/helm-broker/templates/deploy.yaml [hunk 6]
```diff
@@ -135,22 +113,19 @@ spec:
...
-              - curl
-              - localhost:{{ .Values.ctrl.statusPort }}/ready
-          failureThreshold: 3
+          httpGet:
+            path: /ready
+            port: {{ .Values.ctrl.statusPort }}
           periodSeconds: 10
-          successThreshold: 1
           timeoutSeconds: 2
+          initialDelaySeconds: 5
```
Comment (mszostok) [inline resources/helm-broker/templates/deploy.yaml]: in other components you have `10`, maybe it's better to also increase here this delay?

File: tests/service-catalog/Dockerfile [hunk 7]
```diff
@@ -11,6 +11,7 @@ RUN CGO_ENABLED=0 go build -o /env-tester.bin ./cmd/env-tester/main.go
 RUN CGO_ENABLED=0 go test -tags=acceptance -c ./test -o /servicecatalog.test
 
 FROM alpine:3.10
```
Comment (mszostok) [inline tests/service-catalog/Dockerfile]: why it's not migrated to scratch? if cannot be migrated then add comment explaining why

File: components/application-broker/Dockerfile (hunk 3 above)
Comment (polskikiel) [reply components/application-broker/Dockerfile]: removed

File: resources/helm-broker/templates/deploy.yaml (hunk 6 above)
Comment (polskikiel) [reply resources/helm-broker/templates/deploy.yaml]: yes it should be set to 10 👍

File: tests/service-catalog/Dockerfile (hunk 7 above)
Comment (polskikiel) [reply tests/service-catalog/Dockerfile]: added a comment

**Pull request merged**